src/XMakeBuildEngine/UnitTestsPublicOM/Construction/ProjectRootElement_Tests.cs

```diff
@@ -1204,6 +1205,42 @@ public void DeepCloneWithProjectExtensionsElementOfXml()
         }
 
         /// <summary>
+        /// Tests DeepClone and CopyFrom when there is metadata expressed as attributes
+        /// </summary>
+        [Fact]
+        public void DeepCloneWithMetadataAsAttributes()
+        {
+            var project =
+@"<?xml version=`1.0` encoding=`utf-8`?>
+  <Project xmlns = 'msbuildnamespace'>
+
+    <ItemGroup>
+      <Compile Include=`Class1.cs` A=`a` />
+      <Compile Include=`Class2.cs` />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <P>val</P>
+    </PropertyGroup>
+
+    <Target Name=`Build`>
+      <ItemGroup>
+        <A Include=`a` />
+        <B Include=`b` M1=`v1`>
+          <M2>v2</M2>
+        </B>
+        <C Include=`c`/>
+      </ItemGroup>
+    </Target>
+
+  </Project>";
+
+            var pre = ProjectRootElement.Create(XmlReader.Create(new StringReader(ObjectModelHelpers.CleanupFileContents(project))));
+
+            ValidateDeepCloneAndCopyFrom(pre);
```

Comment:
Yup, it had failed with the same `"The element <A> beneath element <Compile> is invalid because a child element with that name already exists"`. The `RemoveAllChilddren` from @jviau's repro did not seem necessary to cause the crash.